README.md

```diff
@@ -1,27 +1,38 @@
-# epm
-
-Ethereum Package Managment
+# EPM
 
 [![Join the chat at https://gitter.im/ethpm/Lobby](https://badges.gitter.im/ethpm/Lobby.svg)](https://gitter.im/ethpm/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-# Goals
-
-The goal of this repository is to create a specification that can be used for
-package management for the Ethereum Virtual Machine (EVM).
+Ethereum Packaging Specification
 
-This explicitely includes:
 
-* Specification for a *manifest* file.
+## Use Cases
 
+The following use cases were used to guide the specification.
 
-This explicitely does not include:
+1. Code with zero deployed instances.
+    1. Reusable base contracts.
+2. Code with one or more addresses.
+    1. Libraries: Installation of the package involves having an importable interface for the library and linking against the proper address during deployment.
+    2. Contracts: Installation of the package involves having an importable interface and some manner of templating in the proper addresses into the local source code prior to compilation as well as other non-contract code that may interact with the installed contract.
 
-* ????
 
+It is worth pointing out that the *Library* and *Contract* use cases are functionally the same.  Once [Solidity issue #242](https://github.com/ethereum/solidity/issues/242) has been addressed this distinction may no longer need to exist.
 
-# Resources
 
-The following resources are encouraged reading before joining the discussion.
+## Open Questions
 
-* https://github.com/purescript/purescript/issues/631
-    * Long discussion on package management from the purescript community.
+* Version
+    * Should the version live in the package manifest, release manifest, or both.
+        * Case for Package
+            * Defines the *current* version of the package.
+        * Case for Release
+            * Each release should have a clearly defined version.
+        * Case for Both
+            * Seems to work well for release workflow. Package manager would
+              populate the release manifest version field from the package
+              manifest version when creating a new release.
+* Validation of the concept of Package Dependencies vs Release Dependencies
+    * When a dependency is defined at the package level it *can* be either pinned to a specific release, or possibly a range of releases such as `^0.4.0` to allow any version in the `0.4.x` line.
```

Comment:
>Package manager uses whatever registrar(s) it is configured to use to resolve all relevant version numbers that match the given semver. (if the value isn't a valid semver it probably just looks for an exact string match).

The problem with this is you're not conflating two different packages. The same package name and version from two different repos is not the same package.

>Package manager picks the greatest version that matches the specified semver number for the dependency.

Although semver helps with this, you're exploding the complexity of verifying the contract graph with this decision. It's better to keep the graph being verified small by defining its nodes as explicitly as possible. We are concerned with validating DAGs of contracts, right?

>I'm not sure I follow your logic at all here because my experience with package indexes is that they keep references to every release that's ever been made. I'm not sure why this would be any different.

For example, FreeBSD does a "ports freeze" before release to insure that the ports tree used for the production release is stable. (https://www.freebsd.org/doc/en_US.ISO8859-1/articles/committers-guide/ports.html#ports-qa-freeze-what)
Given that one of the major goals of this process is to verify that a package will perform as advertised in an adversarial environment, we need to support the ability to provide verified (and versioned) repos. I think we discussed this on gitter or something?